File: C#/136-single-number.cs
public class Solution {
  public int SingleNumber(int[] nums)
        {
            int solution = Int32.MaxValue;
            Dictionary<int, int> numberCounts = new Dictionary<int, int>(); // <num, count>
            for (int i = 0; i < nums.Length; i++)
            {
                if (numberCounts.ContainsKey(nums[i]))
                {
                    numberCounts[nums[i]] += 1;
                }
                else
                {
                    numberCounts.Add(nums[i], 1);
                }
            }
             foreach(int key in numberCounts.Keys)
            {
                if (numberCounts[key] == 1)
                {
                    solution = key;
                }
            }
            return solution;
        }
}

File: Java/136-single-number.java
class Solution {
    // Efficiency: Time: O(n), Space: O(n)
    // Explanation: iterate through the array of integers, storing each integer's count in a map, then finding which integer has a count of 1.
    public int singleNumber(int[] nums) throws IllegalArgumentException {
        HashMap<Integer, Integer> map = new HashMap<>(); // integer, count
        for(int i = 0; i < nums.length; i++) {
            int current = nums[i];
            if(map.containsKey(current)) {
                int count = map.get(current);
                count += 1;
                map.put(current, count);
            } else {
                map.put(current, 1);
            }
        }
        for(Integer key : map.keySet()) {
            int val = map.get(key);
            if(val == 1) {
                return key;
            }
        }
        return -1;
    }
}